src/shogun/classifier/LDA.cpp

```diff
@@ -34,8 +34,8 @@ CLDA::CLDA(
     : CDenseRealDispatch<CLDA, CLinearMachine>(), m_gamma(gamma)
 {
 	init();
-	set_features(traindat);
-	set_labels(trainlab);
+	put("features", dynamic_cast<CFeatures*>(traindat));
```

Comment:
nooooo
I would really try to avoid using this string based setting of parameters inside c++ code. It is more for swig

Reply:
Sorry those are within class methods.... I didnt realise

Reply:
the dynamic cast is unchecked here

Reply:
if it was for me I would make both labels and features read-only (not puttable), but those are only received through a "train" method. But no need to do that here now

Reply:
err I'm a bit confused. Anyway, I just assign them directly now.